drivers/serial/uart_stm32.c

```diff
@@ -685,7 +689,8 @@ static const struct uart_stm32_config uart_stm32_cfg_##name = {		\
 };									\
 									\
 static struct uart_stm32_data uart_stm32_data_##name = {		\
-	.baud_rate = DT_UART_STM32_##name##_BAUD_RATE			\
+	.baud_rate = DT_UART_STM32_##name##_BAUD_RATE,			\
+	.hw_fc = STM32_##name##_HW_FC \
```

Comment:
You could have used CONFIG_##name##_HW_FC directly here and spare the lines in each device declaration below